### DISCUSSION TITLE
Is it possible to change the thrust coefficient, thereby changing the turbine's power?

### DISCUSSION THREAD

**Discussion by Siantos** [NatLabRockies/floris#993] [Q&A] [answered]
In FLORIS, is the power of the turbine calculated by determining the inflow wind speed through wake modeling and then looking up the thrust coefficient table?Is it possible to change the thrust coefficient, thereby changing the turbine's power?If possible, can this be set directly in fmodel.set, or does it require modifying the code files additionally?

**Comment by misi9170** [comment, marked as answer]
Hi @Siantos,

The power is determine using the power table, which is sampled at the inflow wind speed (determined via wake modeling). The thrust coefficient is much the same---it is looked up from a dedicated thrust coefficient table at the inflow wind speed. There is nothing strictly tying these together---that is, the user could build a turbine model with a thrust coefficient table and power table that are not physically realistic, and FLORIS will still run with those.

There is no direct way to control the thrust coefficient, and even if there was, it would not also affect the power due to the separation of the power and thrust coefficient curves described above. However, we _do_ have a method for setting a derating power setpoint for a turbine that is lower than its rated power, and then approximating the thrust coefficient that would result from this derating. This uses the `SimpleDeratingTurbine` operation model. If that is of interest to you, see [documentation here](https://nrel.github.io/floris/operation_models_user.html#simple-derating-model) and [example here](https://github.com/NREL/floris/blob/main/examples/examples_control_types/001_derating_control.py).

For more information on how turbine power and thrust curves are implemented in FLORIS, see the following pages of the documentation:
- [Turbine curves](https://nrel.github.io/floris/turbine_interaction.html)
- [Turbine operation models](https://nrel.github.io/floris/operation_models_user.html)

**Reply by Siantos** [reply]
Thank you for your answer

**Reply by zyla123** [reply]
Hello，Siantos. I am currently facing a similar problem. Can we discuss it with you？
